azerothcore/azerothcore-wotlk
# (Core/creature) Fix pvp tags for mobs

Not sure where to put this, main author of the logic and stuff is https://github.com/neifion-00000000 , I just wrote the code for now to simplify and validate that it could work.

## Changes Proposed:
-  Use the flags "guard" and "civilian" to make NPC pvp targets, as well as some factions by default.
-  Apply "guard" tag where needed for the NPCs that are still missing (eg Astranaar Sentinels)

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/6053

## SOURCE:

Most of the research done is in the file here. But basically it seems the most logical indication of PVP flagging is not the faction alone (which it cannot be) but a combination of guards and civilian. Also we're using PVP factions because as far as we know these are neither guards nor civilians but always PVP flagged.
[faction_setpvp_write_up.txt](https://github.com/azerothcore/azerothcore-wotlk/files/7358044/faction_setpvp_write_up.txt)




## Tests Performed:

- 
- 


## How to Test the Changes:


1. Go to laughing sister: not pvp flagged (.go c 34210)
2. Go to Astranaar sentinel: pvp flagged (.go c 32429) (not yet, once we have decided on the way to go for the flags)
3. Other mobs around the world should now also show the correct flag.

## Known Issues and TODO List:


- [ ] Some mobs need to have a PVP flag added but we haven't figured out which yet. The wiki suggests we could use UNIT_FIELD_BYTES_2, but as far as I can tell that gets read *after* setting the PVP (which is written on the same entry as far as I can tell). 
![image](https://user-images.githubusercontent.com/2592673/137598412-b5a7b07f-fd73-4835-8b4a-3b6ab0a98ab7.png)



## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
